src/Stryker.Core/Stryker.Core/Baseline/DiskBaselineProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Stryker.Core.Logging;
+using Stryker.Core.Options;
+using Stryker.Core.Reporters.Json;
+using System.IO;
+using System.IO.Abstractions;
+using System.Threading.Tasks;
+
+namespace Stryker.Core.Baseline
+{
+    public class DiskBaselineProvider : IBaselineProvider
+    {
+        private readonly StrykerOptions _options;
+        private readonly IFileSystem _fileSystem;
+        private readonly ILogger<DiskBaselineProvider> _logger;
+
+        public DiskBaselineProvider(StrykerOptions options, IFileSystem fileSystem = null)
+        {
+            _options = options;
+            _fileSystem = fileSystem ?? new FileSystem();
+            _logger = ApplicationLogging.LoggerFactory.CreateLogger<DiskBaselineProvider>();
+        }
+
+
+        public async Task<JsonReport> Load(string version)
+        {
+            var reportPath = Path.Combine(_options.BaselineOutputPath, version, "stryker-report.json");
+            try
+            {
+                using StreamReader inputReader = _fileSystem.File.OpenText(reportPath);
+
+                var reportJson = await inputReader.ReadToEndAsync();
+
+                return JsonConvert.DeserializeObject<JsonReport>(reportJson);
+            } 
+            catch(FileNotFoundException)
+            {
+                _logger.LogInformation("No baseline was found at {0}", reportPath.ToString());
+                return null;
+            }
+           
+        }
+
+        public async Task Save(JsonReport report, string version)
+        {
+            var reportPath = Path.Combine(_options.BaselineOutputPath, version);
+
+            var reportJson = report.ToJson();
+
+            _fileSystem.Directory.CreateDirectory(reportPath);
+
+            using StreamWriter outputWriter = _fileSystem.File.CreateText(Path.Combine(reportPath, $"stryker-report.json"));
+
+            await outputWriter.WriteAsync(reportJson);
+
+            _logger.LogDebug($"Baseline report has been writter to {Path.Combine(reportPath, $"stryker-report.json")}");
```

Comment:
Also add an informational variant of this logging that simply says the baseline has been saved. Or remove that same logging from the azure file share provider so they are consistent in their informational logging.